src/app/veilederverktoy/begrunnelseform/maltekster-form.tsx

```diff
@@ -18,29 +18,44 @@ type MalteksterFormProps<T extends BegrunnelseValues> = OwnProps<T> & InjectedIn
 
 function MalteksterForm(props: MalteksterFormProps<any>) {
 
-    function onChange(e: ChangeEvent<HTMLSelectElement>) {
-        if (e.target.value) {
-            const message = props.intl.formatMessage({ id: 'innstillinger.modal.start-eskalering-' + e.target.value});
-            context.setFieldValue('begrunnelse', message);
-            context.setFieldValue('tekst', message);
-        } else {
-            context.resetForm();
+    function Maltekstvelger(maltekstvelgerProps: any) {
```

Comment:
Hvorfor ha en funksjon i en funksjon?
Nå vil denne funksjonen bli laget på nytt hver gang funksjonen den er inni bli kaldt. Er litt ueffektivt. Flytt helst ut i egen fil

Reply:
Any typen er også litt NO NO. Bedre å sette de faktiske typene man bruker så man får hjelp av type systemet til typescript

Reply:
Funksjon i funksjon fordi jeg antok at det er slik scope funker (som i f.eks. Java). Ønsket var å begrense scope, i det minste med tanke på å lese intensjonen i koden.

Skal se på å flytte den i egen fil, noe jeg ser på som en svakhet i språket. Da 1) får jeg enda en ny fil, bare for å holde en funksjon, og 2) jeg mister lesbarheten som her sterkt antyder at den skal brukes internt i `maltekster-form.tsx` og bare der.

`any` ville jeg ikke brukt dersom det var en annen type som umiddelbart (uten hjelp fra Typescript/IntelliJ) falt meg inn, men siden jeg ble satt under såpass dårlig tid at jeg også gikk glipp av en legetime så ble det `any`. Du foreslår ikke en spesifikk type i stedet?

Reply:
Helst aldri bruk funksjon i funksjon. Kan gjerne ha 2 funksjoner i samme fil om du ønsker å scope bruken til den ene funksjonen til den andre. Det er bare å ikke exporte den funksjonen

Typen du ønsker i stedet for `any` er noe ala
```tsx
    function Maltekstvelger(maltekstvelgerProps: {visible: boolean})
```